### PR TITLE
Copter: Added autorotation flight mode page

### PR DESCRIPTION
**Pull request by Gone4Dirt**
This PR addresses this issue #2388 

A page for the autorotation flight mode has been added.  As this flight mode is still under development the setup instructions are not yet written on this wiki page.  Instead, the discourse thread has been linked for the latest instructions.  This was the approach agreed upon in a dev call some time ago.  Once the mode is available for real world flight testing I will update this page with the instructions.

### CODE REVIEW

File: copter/source/docs/flight-modes.rst [hunk 1]
```diff
@@ -45,6 +45,7 @@ companion computer.
    <tr><td>SysID</td><td>-</td><td>+</td><td></td><td>Special diagnostic/modeling mode</td></tr>
    <tr><td>Throw</td><td>A</td><td>A</td><td>Y</td><td>Holds position after a throwing takeoff</td></tr>
    <tr><td>ZigZag</td><td>A</td><td>A</td><td>Y</td><td>Useful for crop spraying</td></tr>
+   <tr><td>Autonomous Autorotation</td><td>A</td><td>A</td><td>Y</td><td>Used for emergencies in traditional helicopters. Helicopter only.  Currently SITL only.</td></tr>
```
Comment (Hwurzburg) [inline copter/source/docs/flight-modes.rst]: placement should be alphabetical in the lists...

File: copter/source/docs/traditional-helicopter-autorotation-mode.rst [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+the autorotation manoeuvre, in which the descending aircraft can use the airflow being forced up through the main blades to
+drive the rotor and still produce some upwards thrust.  The primary goal of the manoeuvre is to use the aircrafts remaining energy 
+to bring the aircraft safely to the ground, landing softly.
+
+The manoeuvre can be broken down into phases.  The breakdown and description of those phases differs across the literature that 
+is available on the topic.  The description that is given here matches the approach taken in the implementation with ArduCopter.
+
+There are four phases of the autorotation:
+
+**1. Entry** - The initial actions of the vehicle in which collective is used to minimise head speed loss and cyclic is used for attitude control
```
Comment (Hwurzburg) [inline copter/source/docs/traditional-helicopter-autorotation-mode.rst]: minimize

File: copter/source/docs/traditional-helicopter-autorotation-mode.rst [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+==========
+
+The autonomous autorotation flight mode cannot be entered via a mode switch, as is done with other flight modes.  In an attempt to mimic the autorotation procedure 
+followed when flying a manual autorotation, the mode is entered and exited using the motor interlock switch.  The autonomous flight mode can only be entered if 
+the interlock is disengaged whilst flying in a non-manual collective mode.  When exiting the autorotation the flight mode is returned to the mode from which it came 
+(e.g. Altitude Hold -> Autonomous Autorotation -> Altitude Hold).  The only exception to this is when the aircraft enters the autorotation from an auto mission.  
+In which case, the mode is exited into Altitude Hold.  This is done to prevent the aircraft from potentially resuming its auto mission and flying further away 
+from home, whereby in the scenario it has autonomously recovered from a power failure and is therefore high risk for another power failure.
+
+The diagram below illustrates the mode switchology:
```
Comment (Hwurzburg) [inline copter/source/docs/traditional-helicopter-autorotation-mode.rst]: learned a new word today!....perhaps "mode switching scheme" would be more straightforward than switchology?  your choice

**Comment by Hwurzburg** [comment]
Really superb first pass...love the diagrams !(if only motor interlock use and effects were explained somewhere in the wiki...I must look at the videos and write something up on that)

File: copter/source/docs/traditional-helicopter-autorotation-mode.rst (hunk 3 above)
Comment (Gone4Dirt) [reply copter/source/docs/traditional-helicopter-autorotation-mode.rst]: Bill and I discussed this at length a while back.  I wasn't sure if it was a real word at the time but it is the correct technical term to use.  I think it is straight forward enough to infer its meaning as well.

**Comment by Gone4Dirt** [comment]
Thanks for the review @Hwurzburg.  I will make the changes you have requested.
With regards to the interlock, I am happy to write a page on that two if you would like?  I would just need some guidance as to where it should go in the Wiki.

File: copter/source/docs/flight-modes.rst (hunk 1 above)
Comment (Gone4Dirt) [reply copter/source/docs/flight-modes.rst]: Moved the autorotation mode to be in its alphabetical position in all lists on this page.

File: copter/source/docs/traditional-helicopter-autorotation-mode.rst (hunk 2 above)
Comment (Gone4Dirt) [reply copter/source/docs/traditional-helicopter-autorotation-mode.rst]: Changed to US spelling.

**Comment by Gone4Dirt** [comment]
I have made those changes and pushed them.

**Comment by Hwurzburg** [comment]
LGTM...merging thanks!

**Comment by rmackay9** [comment]
Yes, a really good page, thanks!

**Pull request merged**